findDate.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Searches for dates in an array of strings. The dates can be in form
+ * dd.mm.yyyy, dd.mm.yy or yyyy-mm-dd
+ *
+ * @param {string[]} fragments array of string fragments to be scanned for dates
+ * @returns the date in the input array as js Date object or null, if no strings contain a date. If multiple dates are found, returns the date closest to today.
+ */
+function findDate(fragments) {
+  let ret = null;
+  let matches = [];
+  for (let i = 0; i < fragments.length; i += 1) {
+    const m1 = fragments[i].match(
```

Comment:
1. Checks m1 and m2 will result in adding wrong dates to matches array. For dd.mm.yyyy format, m2 will push wrong date in matches (eg "01.01.2012 will be pushed as "01.01.20"). To avoid this, checks for date formats dd.mm.yyyy and dd.mm.yy can be combined into one. 
```suggestion
    const m1 = fragments[i].match(/[0-9][0-9]\.[0-9][0-9]\.(\d{2}|\d{4})+/g);
```
2. dates can already be converted to date objects using new Date() before pushing to matches. In this way, conversion in multiple lines (47-51) can be avoided. Also, it will be useful in further validation of  input dates.

3. Pseudo code for the flow:
- Validate date format
- If valid, convert to date object and push to matches
- Remove invalid dates from matches array
- Check for closest day from today. Add a check to ignore future dates and dates older than 10 years
- Return JS date object or null (if no strings contain a date)